frontend/ui/widget/configdialog.lua

```diff
@@ -674,8 +670,16 @@ function ConfigOption:_itemGroupToLayoutLine(option_items_group)
     local j = self.config.panel_index
     for i, v in ipairs(option_items_group) do
         if v.name then
-            layout_line[j] = v
-            j = j + 1
+            if type(v.layout) == "table" and type(v.disableFocusManagement) == "function" then
```

Comment:
If `v.layout` can only be a table or `nil`, and if `v.disableFocusManagement` can only be a function or `nil`, no need for type() (which we rarely see in our code): just use `if v.layout and v.disableFocusManagement then`.

Reply:
Yes!

Most of time, I work with strong typed language. lua such dynamic language make me cautious.

`ConfigDialog` is a big container, using `type` can ensure the `v` is not nil and make it more possibility is a `FocusManager` needed here.

Reply:
> using type can ensure the v is not nil 

```
> v=nil
> if type(v.layout) == "table" then print("ok") end
  stdin:1: attempt to index global 'v' (a nil value)
```
As Lua is not strong typed, we expect to get crashes when coding to signal us of our errors :)
`if v.layout` will crash if v is nil, which might be fine
if v can be nil, you need to use:
`if v and v.layout`

Seeing `if type(v.layout) == "table" then ` can make us and futur developpers think that v.layout can be someething else than a table, and I would go looking for it.

The code can document a bit the context :)
(Just explaning how I/we apprehend Lua code in KOReader, not complaining at all about your rather realy good coding style :)